scene/gui/rich_text_label.cpp

```diff
@@ -3196,19 +3198,21 @@ bool RichTextLabel::remove_paragraph(const int p_paragraph) {
 	}
 
 	// Remove all subitems with the same line as that provided.
-	Vector<int> subitem_indices_to_remove;
-	for (int i = 0; i < current->subitems.size(); i++) {
-		if (current->subitems[i]->line == p_paragraph) {
-			subitem_indices_to_remove.push_back(i);
+	Vector<List<Item *>::Element *> subitem_to_remove;
+	for (List<Item *>::Element *E = current->subitems.front(); current_frame->lines[p_paragraph].from->E; E = E->next()) {
+		if (E->get()->line == p_paragraph) {
+			subitem_to_remove.push_back(E);
+		} else {
+			break;
 		}
 	}
```

Comment:
I guess this could be simplified like so?

```suggestion
	for (const Item *E : current->subitems) {
		if (!current_frame->lines[p_paragraph].from->E || E->line != p_paragraph) {
			break;
		}

		subitem_to_remove.push_back(E);
	}
```

And below we would have `Item *` instead of `List<Item *>::Element`, removing all the `->get()`s.

The same cannot be done to the first for-loop, as it doesn't start at the front, and instead starts at `current_frame->lines[p_line].from->E`, unfortunately.

Reply:
Actually, it should start from `current_frame->lines[p_line].from->E` as well.

Reply:
Updated.